extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -418,9 +421,11 @@ export class SchemaCompareResult {
 		}).component();
 
 		this.generateScriptButton.onDidClick(async (click) => {
+			Telemetry.sendTelemetryEvent('SchemaCompareGenerateScript');
 			let service = await SchemaCompareResult.getService('MSSQL');
 			let result = await service.schemaCompareGenerateScript(this.comparisonResult.operationId, this.targetEndpointInfo.serverName, this.targetEndpointInfo.databaseName, azdata.TaskExecutionMode.script);
 			if (!result || !result.success) {
+				Telemetry.sendTelemetryEvent('SchemaCompareGenerateScriptFailed');
```

Comment:
>SchemaCompareGenerateScriptFailed [](start = 34, length = 33)

For all these failures - are there a certain subset of failures that you expect to be sent the majority of the time that you bucket the error message on? For example, let's say that one of the errors returned could be 'Could not connect to Server 'MyServer'.' 

Then here you could have a helper method with like

getTelemetryErrorMessage(msg: string) {
   if(msg.Contains('Could not connect to Server') return 'CouldNotConnectToServer';
   else if (....)
   ....
   else return 'Other';
}

That way you could see more useful information about the failures.

Reply:
Agreed with this, we should definitely have data to support "why it failed"

Reply:
Most of the actual failure messages show in the task view. The only error I know of that would come here is if the database can't be connected to, so I added an error type for that.